DesignSafe-CI/portal
# Task/DES-1988: Add hazmapperMaps field to project metadata and display in project preview views.

## Overview: ##
Adds the `hazmapperMaps` field to the Project models and displays it in the project views.
The fields are intended to be directly set/updated from Hazmapper (Will later be able to test with #911 and TACC-Cloud/hazmapper#70 [WIP]).

## PR Status: ##


## Related Jira tickets: ##

* [DES-1988](https://jira.tacc.utexas.edu/browse/DES-1988)

## Summary of Changes: ##
Modified the Project-related models for django and elasticsearch to include the `hazmapperMaps` field, which is looks like (List of objects):
```
[{
  name: string (name of the map: given in hazmapper),
  path: string (),
  deployment: string (),
  uuid: string ()
}]
```

Also modified project views to include the field values.

## Testing Steps: ##
1. Create a maps in [hazmapper prod](https://hazmapper.tacc.utexas.edu/hazmapper/) and [hazmapper staging](https://hazmapper.tacc.utexas.edu/staging/)
2. Try making a request to the `api/projects/$UUID` either by adding a request value in the `update()` function of `manage-project.component.js`) or with postman/curl with #911 and the body :
```
{
    "uuid": $UUID,
    "hazmapperMaps": [
        {
            "name": $STAGING_MAP_NAME, 
            "path": $STAGING_MAP_PATH, 
            "uuid": $STAGING_MAP_UUID,
            "deployment": "staging"
        },
        {
            "name": $PROD_MAP_NAME, 
            "path": $PROD_MAP_NAME, 
            "uuid": $PROD_MAP_NAME,
            "deployment": "production"
        }
    ]
}
```
3. Ensure that the project-view of the project properly displays the list of hazmapper maps.
4. Ensure that the external link to hazmapper opens to the proper hazmapper map with the matching uuid and deployment.

## UI Photos:
![Screenshot from 2021-11-09 17-47-05](https://user-images.githubusercontent.com/9425579/141023683-c8b808c3-ad29-44eb-a0e8-a1f130bff58d.png)

## Notes: ##
Please let me know if I am missing something in regards to updating the project model's fields and project views.
There is also a lot of redundant code that I wasn't sure how to split at the moment (especially for the different project views). So, I just kept how it is now (similar to `onBrowse()`).
